zerver/tests/test_subs.py

```diff
@@ -2269,40 +2269,40 @@ def test_subscriptions_add_stream_with_null(self) -> None:
                                "Stream name '%s' contains NULL (0x00) characters." % (stream_name,))
 
     def test_user_settings_for_adding_streams(self) -> None:
-        with mock.patch('zerver.models.UserProfile.can_create_streams', return_value=False):
+        with mock.patch('zerver.models.UserProfile.has_permission', return_value=False):
             result = self.common_subscribe_to_streams(self.test_user, ['stream1'])
             self.assert_json_error(result, 'User cannot create streams.')
 
-        with mock.patch('zerver.models.UserProfile.can_create_streams', return_value=True):
+        with mock.patch('zerver.models.UserProfile.has_permission', return_value=True):
             result = self.common_subscribe_to_streams(self.test_user, ['stream2'])
             self.assert_json_success(result)
 
         # User should still be able to subscribe to an existing stream
-        with mock.patch('zerver.models.UserProfile.can_create_streams', return_value=False):
+        with mock.patch('zerver.models.UserProfile.has_permission', return_value=False):
             result = self.common_subscribe_to_streams(self.test_user, ['stream2'])
             self.assert_json_success(result)
 
-    def test_can_create_streams(self) -> None:
+    def test_has_permission(self) -> None:
         othello = self.example_user('othello')
         othello.role = UserProfile.ROLE_REALM_ADMINISTRATOR
-        self.assertTrue(othello.can_create_streams())
+        self.assertTrue(othello.has_permission('create_stream_policy'))
```

Comment:
@timabbott are tests fine using has_permission or I should keep `can_create_streams` as you suggested above to keep it as wrappers?

Reply:
I think `can_create_streams` might be more readable here, but either way is probably fine in tests.